module/app/app.go

```diff
@@ -696,9 +727,18 @@ func (app *Gravity) Name() string { return app.BaseApp.Name() }
 
 // BeginBlocker application updates every begin block
 func (app *Gravity) BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock) abci.ResponseBeginBlock {
+	firstBlock.Do(func() { // Run the startup firstBeginBlocker assertions only once
```

Comment:
this firstBlock construct is pretty cool. Anything else we want to consider asserting? Invariants where on the table but that might not be good, if an invariant is violated mid run the user would not be able to restart their node.